src/skmultiflow/trees/multi_target_regression_hoeffding_tree.py

```diff
@@ -420,8 +443,13 @@ def predict(self, X):
         list
             Predicted target values.
         """
-        r, _ = get_dimensions(X)
 
+        if not self._n_targets_set:
+            raise RuntimeError(
```

Comment:
I agree with you. Do you have any suggestions? I think that we could simply return `0`, but I'm not sure how this action would impact the upper layers, such as Evaluators and Metric Collections.

Reply:
The binary and multi-class classification evaluators should work without issue. They are designed to support emerging classes. This should be the case as well for the regression case. I agree that we should just return `[0]`. I expect that we will update the regression evaluators in future releases to improve performance and this will be part of the design criteria.

Reply:
I changed the behavior of the MTR HTs to return zero if `predict` is called before the first `partial_fit` in commit [384ef1b](https://github.com/scikit-multiflow/scikit-multiflow/pull/168/commits/384ef1bac5731aaf46726d78997f351f85a1aac4). The tests were also updated to cover this situation.